src/persist/src/location.rs

```diff
@@ -949,21 +956,24 @@ pub mod tests {
         // expected is correct.
         assert_eq!(
             consensus
-                .compare_and_set(deadline, Some(state.seqno), invalid_regressing_seqno)
+                .compare_and_set(key, deadline, Some(state.seqno), invalid_regressing_seqno)
                 .await,
             Err(ExternalError::from(anyhow!("new seqno must be strictly greater than expected. Got new: SeqNo(3) expected: SeqNo(5)")))
         );
 
         // Can correctly update to a new state if we provide the right expected seqno
         assert_eq!(
             consensus
-                .compare_and_set(deadline, Some(state.seqno), new_state.clone())
+                .compare_and_set(key, deadline, Some(state.seqno), new_state.clone())
                 .await,
             Ok(Ok(()))
         );
 
         // We can observe the a recent value on successful update.
-        assert_eq!(consensus.head(deadline).await, Ok(Some(new_state.clone())));
+        assert_eq!(
+            consensus.head(key, deadline).await,
+            Ok(Some(new_state.clone()))
+        );
```

Comment:
let's either add some test coverage for different keys to this or a TODO about it

Reply:
leave it as a todo I can add more tests